tests/integration/test_charm.py

```diff
@@ -1,49 +1,137 @@
-    assert ops_test.model
-    charm = await ops_test.build_charm(".")
-    await ops_test.model.deploy("postgresql-k8s")
-    await ops_test.model.deploy(charm)
-    await ops_test.model.wait_for_idle(raise_on_error=False)
-    await ops_test.model.add_relation(
-        "postgresql-k8s:db",
-        "mattermost-k8s",
+    # create a team
+    data = {"name": "test", "display_name": "test", "type": "O"}
+    response = requests.post(
+        f"http://{mattermost_ip}:8065/api/v4/teams", data=json.dumps(data), headers=headers
     )
-    # mypy doesn't see that ActiveStatus has a name
-    await ops_test.model.wait_for_idle(status=ActiveStatus.name, raise_on_error=False)  # type: ignore
+    team = response.json()
 
+    # create a channel
+    data = {"team_id": team["id"], "name": "test", "display_name": "test", "type": "O"}
+    response = requests.post(
+        f"http://{mattermost_ip}:8065/api/v4/channels", data=json.dumps(data), headers=headers
+    )
+    channel = response.json()
 
-async def test_status(ops_test: OpsTest):
-    assert ops_test.model
-    # mypy doesn't see that ActiveStatus has a name
-    assert ops_test.model.applications["mattermost-k8s"].status == ActiveStatus.name  # type: ignore
-    assert ops_test.model.applications["postgresql-k8s"].status == ActiveStatus.name  # type: ignore
+    # create a test file
+    test_file = tmp_path / "test_file.txt"
+    test_content = "This is a test file."
+    test_file.write_text(test_content, encoding="utf-8")
 
+    # upload the test file
+    await ops_test.juju("run", "--application", app.name, cmd)
+    with open(test_file, "r", encoding="utf-8") as test_file:
+        response = requests.post(
+            f"http://{mattermost_ip}:8065/api/v4/files",
+            data={"channel_id": channel["id"]},
+            files={"file": test_file},
+            headers=headers,
+        )
 
-async def test_workload_online_default(ops_test: OpsTest):
-    assert ops_test.model
-    app = ops_test.model.applications["mattermost-k8s"]
-    mmost_unit = app.units[0]
-    action = await mmost_unit.run("unit-get private-address")
-    curl_output = await juju_run(
-        mmost_unit, "curl {}:8065".format(action.results["Stdout"].replace("\n", ""))
+    logger.info("Mattermost config updated, checking bucket content")
+
+    # Configuration for boto client
+    s3_client_config = Config(
+        region_name=localstack_s3_config["region"],
+        s3={
+            "addressing_style": "virtual",
+        },
     )
-    assert "Mattermost" in curl_output
+
+    # Configure the boto client
+    s3_client = client(
+        "s3",
+        localstack_s3_config["region"],
+        aws_access_key_id=localstack_s3_config["credentials"]["access-key"],
+        aws_secret_access_key=localstack_s3_config["credentials"]["secret-key"],
+        endpoint_url=localstack_s3_config["endpoint"],
+        use_ssl=False,
+        config=s3_client_config,
+    )
```

Comment:
Out of curiosity, would this be better suited as a fixture? I don't think it requires any runtime dependencies at the moment :)

Reply:
Moved to a fixture